lib/pbench/cli/server/shell.py

```diff
@@ -1,14 +1,18 @@
 #!/bin/env python3
 
-from pbench.server.api import create_app
+from pbench.server.api import create_app, get_server_config
 import subprocess
 
 
+server_config = get_server_config()
+application = create_app(server_config)
```

Comment:
Seems like these should not be global variables but local only to `main()`.

Reply:
I had to do this because the `gunicorn` expect the format to be `{module}:flask_app_instance` to run the app. If I define it inside the main function then I have to do something like `pbench.cli.server.shell:main()` where `main` function need to return the flask_app instance. Another alternative would be to create another file say `wsgi.py` to hold the application and `shell.py` would import it and then gunicorn can call `pbench.cli.server.wsgi:application()` where application function in `wsgi.py` would return the app instance

Reply:
Right, but why not declare only `application` as `global`, but set its value to `None`.  Then in `main()`, fetch the server configuration, create the application instance, assigning it to the `application` global variable, and then invoke `gunicorn`?

This prevents the application and configuration from being loaded at module load time, so that we can unit test on demand, clearing out the value of the global variable only.

Reply:
Actually while testing now, I realized this would break. We run the gunicorn using the python subprocess and thus it create a separate gunicorn process and for that process the `application` would still remain `None` because it wont pick the same instance of application that we tried to set in the `main`